catalogs/catalog-hadoop/src/main/java/org/apache/gravitino/catalog/hadoop/authentication/UserContext.java

```diff
@@ -0,0 +1,163 @@
+import org.apache.hadoop.security.UserGroupInformation;
+
+public abstract class UserContext implements Closeable {
+
+  private static final Map<NameIdentifier, UserContext> userContextMap = Maps.newConcurrentMap();
+
+  private static void addUserContext(NameIdentifier nameIdentifier, UserContext userContext) {
+    userContextMap.put(nameIdentifier, userContext);
+  }
+
+  public static void clearUserContext(NameIdentifier nameIdentifier) {
+    UserContext userContext = userContextMap.remove(nameIdentifier);
+    if (userContext != null) {
+      try {
+        userContext.close();
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to close user context", e);
+      }
+    }
+  }
+
+  public static void cleanAllUserContext() {
+    userContextMap
+        .values()
+        .forEach(
+            userContext -> {
+              try {
+                userContext.close();
+              } catch (IOException e) {
+                throw new RuntimeException("Failed to close user context", e);
+              }
+            });
+    userContextMap.clear();
+  }
+
+  public static UserContext getUserContext(
+      NameIdentifier nameIdentifier,
+      Map<String, String> properties,
+      Configuration configuration,
+      CatalogInfo catalogInfo) {
+    // Try to get the parent user context.
+    NameIdentifier currentNameIdentifier = NameIdentifier.of(nameIdentifier.namespace().levels());
+    UserContext parentContext = null;
+    while (!currentNameIdentifier.namespace().isEmpty()) {
+      if (userContextMap.containsKey(currentNameIdentifier)) {
+        parentContext = userContextMap.get(currentNameIdentifier);
+        break;
+      }
+      currentNameIdentifier = NameIdentifier.of(currentNameIdentifier.namespace().levels());
+    }
+
+    if (configuration == null) {
+      configuration = new Configuration();
+    }
+    AuthenticationConfig authenticationConfig = new AuthenticationConfig(properties);
+
+    // If we do not set the impersonation, we will use the parent context;
+    boolean enableUserImpersonation = false;
+    if (properties.containsKey(IMPERSONATION_ENABLE_KEY)) {
+      enableUserImpersonation = authenticationConfig.isImpersonationEnabled();
+    } else if (parentContext != null) {
+      enableUserImpersonation = parentContext.enableUserImpersonation();
+    }
+
+    AuthenticationType authenticationType = AuthenticationType.SIMPLE;
+    if (properties.containsKey(AuthenticationConfig.AUTH_TYPE_KEY)) {
+      authenticationType =
+          authenticationConfig.isSimpleAuth()
+              ? AuthenticationType.SIMPLE
+              : AuthenticationType.KERBEROS;
+    }
+
+    UserGroupInformation currentUser;
+    try {
+      currentUser = UserGroupInformation.getCurrentUser();
+    } catch (IOException ioException) {
+      throw new RuntimeException("Failed to get current user", ioException);
+    }
+
+    if (authenticationType == AuthenticationType.SIMPLE) {
```

Comment:
It seems clearer.
```
if (authenticationType == AuthenticationType.SIMPLE) {
} else if (authenticationType == AuthenticationType.KERBEROS) {
} else {
}
```

Reply:
Currently, there are only two types: `AuthenticationType.SIMPLE` and `AuthenticationType.SIMPLE` and IDE does not suggestion to user if/elseif block.

Reply:
If we add another type, it will modify less code.

Reply:
done